# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/ipc/PRemoteDecoderManager.ipdl
# Commit: 57607f8b6d8e
# Full Hash: 57607f8b6d8e1619b5c488494fd57547944c47b1
# Author: Dan Glastonbury <dan.glastonbury@gmail.com>
# Date: 2020-10-21 09:51:28
# Regressor Bug: 1595994
# File Overlap Count: 1
# Description:
#   Bug 1595994 - P7: Add Supports messages to PRemoteDecoderManager. r=kamidphish,nika,ipc-reviewers
#   
#   Add a synchronous Supports message to the IPDL PRemoteDecoderManager protocol so
#   decoder modules can query for playback support in the actual process that will
#   attempt to do the decoding.
# ==============================================================================

diff -r 1932f2f34993 -r 57607f8b6d8e dom/media/ipc/PRemoteDecoderManager.ipdl
--- a/dom/media/ipc/PRemoteDecoderManager.ipdl	Tue Oct 20 23:26:25 2020 +0000
+++ b/dom/media/ipc/PRemoteDecoderManager.ipdl	Tue Oct 20 23:26:27 2020 +0000
@@ -12,6 +12,7 @@
 using AudioInfo from "MediaInfo.h";
 using struct mozilla::layers::TextureFactoryIdentifier from "mozilla/layers/CompositorTypes.h";
 using mozilla::CreateDecoderParams::OptionSet from "PlatformDecoderModule.h";
+using mozilla::DecoderDoctorDiagnostics from "DecoderDoctorDiagnostics.h";
 
 namespace mozilla {
 
@@ -32,6 +33,11 @@
   manages PRemoteDecoder;
 
 parent:
+  sync Supports(RemoteDecoderInfoIPDL info,
+                TextureFactoryIdentifier? identifier)
+         returns (bool success,
+                  DecoderDoctorDiagnostics diagnostics);
+
   sync PRemoteDecoder(RemoteDecoderInfoIPDL info,
                       OptionSet options,
                       TextureFactoryIdentifier? identifier)